### PR TITLE
gui, refactor: Register Qt meta types in application constructor

### PR DESCRIPTION
**Pull request by promag**
Removes a warning when running `QT_QPA_PLATFORM=cocoa src/qt/test/test_bitcoin-qt`.

### CODE REVIEW

**Comment by promag** [comment]
Pushed in #17457, then #19033, and now on it's own PR.

**Comment by jonasschnelli** [comment]
We could share those registrations with the main GUI code as also commented [here](https://github.com/bitcoin/bitcoin/pull/19011/files#r429144851). Since its test only,... not really necessary but would prevent further testing issues.

utACK 41c3af9fecd52d44fb205edec7c0f054036d41eb

**Comment by promag** [comment]
@jonasschnelli yes, was already suggested by @ryanofsky somewhere and I've suggested adding `GUIUtils::registerMetaTypes`.

**Comment by promag** [comment]
Moving to `GUIUtil::registerMetaTypes`, and because of `WalletModel*` registration, I need to `#include <qt/walletmodel.h>` which causes
```
A new circular dependency in the form of "qt/guiutil -> qt/walletmodel -> qt/guiutil" appears to have been introduced.
A new circular dependency in the form of "qt/clientmodel -> qt/guiutil -> qt/walletmodel -> qt/clientmodel" appears to have been introduced.
A new circular dependency in the form of "qt/guiutil -> qt/walletmodel -> qt/optionsmodel -> qt/guiutil" appears to have been introduced.
A new circular dependency in the form of "qt/guiutil -> qt/walletmodel -> qt/paymentserver -> qt/guiutil" appears to have been introduced.
```

**Comment by ryanofsky** [comment]
re: https://github.com/bitcoin/bitcoin/pull/19104#issuecomment-635897334

Slight preference for just adding a new file like src/qt/types.h or src/qt/util_types.h to adding new circular dependencies here

**Comment by promag** [comment]
How about `BitcoinApplication::registerMetaTypes`?

**Comment by ryanofsky** [comment]
> How about `BitcoinApplication::registerMetaTypes` - maybe called from constructor?

Seems good

**Comment by promag** [comment]
Updated, almost move only.

**Comment by jonasschnelli** [comment]
utACK 9df2528e57ff3c22f3d5777eb6a60a326c909995

**Comment by hebasto** [comment]
Concept ACK.

File: src/qt/bitcoin.cpp [hunk 1]
```diff
@@ -253,6 +254,23 @@ bool BitcoinApplication::baseInitialize()
     return m_node.baseInitialize();
 }
 
+void BitcoinApplication::registerMetaTypes()
```
Comment (hebasto) [inline src/qt/bitcoin.cpp:257]: Why a member function?
Could it be a non-member function, either in unnamed namespace or `static`?

**Review by hebasto — CHANGES_REQUESTED** [review]
Tested 9df2528e57ff3c22f3d5777eb6a60a326c909995 on macOS 10.15.5 -- works as expected.

File: src/qt/bitcoin.cpp (hunk 1 above)
Comment (promag) [reply src/qt/bitcoin.cpp]: Good question 🤷  updated.

**Review by hebasto — APPROVED** [review]
ACK 4f49d5222eca11c149713ad34113d5a3d1c577b1, tested on macOS 10.15.5.

**Comment by jonasschnelli** [comment]
Re utACK 4f49d5222eca11c149713ad34113d5a3d1c577b1

**Pull request merged**